pkg/querier/queryrange/splitters.go

```diff
@@ -91,42 +91,67 @@ func (s *defaultSplitter) split(execTime time.Time, tenantIDs []string, req quer
 	}
 
 	var (
-		ingesterSplits []queryrangebase.Request
-		origStart      = req.GetStart().UTC()
-		origEnd        = req.GetEnd().UTC()
+		splitsBeforeRebound []queryrangebase.Request
+		origStart           = req.GetStart().UTC()
+		origEnd             = req.GetEnd().UTC()
+		start, end          = origStart, origEnd
+
+		reboundOrigQuery           bool
+		splitIntervalBeforeRebound time.Duration
 	)
 
-	start, end, needsIngesterSplits := ingesterQueryBounds(execTime, s.iqo, req)
+	switch req.(type) {
+	case *LokiSeriesRequest, *LabelRequest:
```

Comment:
curious why aren't we considering `split_ingester_queries_by_interval` for series and labels queries?

Reply:
i feel these features have orthogonal goals.

`split_ingester_queries_by_interval` is introduced to reduce the number of sub-queries we send to ingesters. currently metadata queries send atmost 1 subquery to ingesters with the default split interval.
 
while `split_recent_metadata_queries_by_interval` would increase the number of sub-queries, they should ideally deamplify over time as the results get cached.

 I am not sure if we'd use these two together for metadata queries and what the config would look like if we do so.

Reply:
Make sense. can you record this somewhere as comment for the future reference?